### PR TITLE
[ALL CHARTS] signing an image with digest instead of tag

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

Based on the comment in [here](https://github.com/prometheus-community/helm-charts/pull/2631). 

#### What this PR does / why we need it

it'd be better to sign images with its digests 

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [ ] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Review by monotek — APPROVED** [review]

**Review by dotdc — DISMISSED** [review]
LGTM

**Comment by monotek** [comment]
@scottrigby 
please have a look :)

**Comment by scottrigby** [comment]
@developer-guy thanks for this PR!

I initially signed by tag for ease, and checked with @sabre1041 who said this is a perfectly valid way to do it, and that it's fine to ignore this cosign CLI warning.

That said... I like your approach using stdout from helm push to get the digest. This is on my radar to check if that is the correct digest to use (helm artifact also has a config digest as well as layer digests).

I believe the underlying issue (CI errors) is now fixed, but this could be. A nice to have if it all works correctly using this helm-returned digest for co-sign keyless signing in the transparency log.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -61,10 +61,12 @@ jobs:
...
-            helm push "${pkg}" oci://ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts
+            helm push "${pkg}" oci://ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts > .digest
             file=${pkg##*/}
             name=${file%-*}
             version=${file%.*}
             version=${version#*-}
-            cosign sign ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts/"${name}":"${version}"
+            digest=$(cat .digest | awk -F "[, ]+" '/Digest/{print $NF}')
+            echo "[+] Signing an image ${name}:${version} with $digest"
+            cosign sign ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts/"${name}":"${digest}"
```
Comment (scottrigby) [inline .github/workflows/release.yaml:71]: Minor fix:
```suggestion
            cosign sign ghcr.io/"${GITHUB_REPOSITORY_OWNER}"/charts/"${name}"@"${digest}"
```

**Review by scottrigby — CHANGES_REQUESTED** [review]
Suggested one fix below. Also PR branch needs to be rebased against updates in main

**Comment by scottrigby** [comment]
Also while you're at it, would you also please change the step name:
```diff
-      - name: Push charts to GHCR
+      - name: Push charts to GHCR and sign
```

**Comment by scottrigby** [comment]
Great! One more thing. There's this linting error (see https://github.com/prometheus-community/helm-charts/actions/runs/3422907447/jobs/5722843306):

> 2022-11-10 00:36:59 [ERROR]   Error code: 1. Command output:
> ------
> .github/workflows/release.yaml:58:9: shellcheck reported issue in this script: SC2002:style:11:16: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead [shellcheck]

**Review by scottrigby — APPROVED** [review]
All looks good! Thanks @developer-guy 😄

**Pull request merged**